### PR TITLE
Update sabre 2.1

### PR DESCRIPTION
**Pull request by PVince81**
Depends on core's sabre 2.1 update: https://github.com/owncloud/core/pull/14151

### CODE REVIEW

**Comment by PVince81** [comment]
Also I'm not 100% sure whether those lines must be removed or replaced with others.
I thought that the new Sabre was automatically mapping these to internal string properties that have all the necessary fixes.

Note that StringProperty and CompoundProperty were removed in core, see https://github.com/owncloud/core/pull/14151#discussion-diff-24604921 and https://github.com/owncloud/core/pull/14151#discussion-diff-24604937

**Comment by PVince81** [comment]
Might help: http://sabre.io/vobject/upgrade/
But not sure what the `$classMap` replacement is.

**Comment by PVince81** [comment]
- [x] import parsing broken: if the vcf contains two groups "group1,group2" it will create a group called "group1,group2" ...
- [x] fix unit test with parse escaping

**Comment by PVince81** [comment]
Obviously the travis tests will not pass until the sabre upgrade is merged in core, as it's testing against master.

Here are the failing local unit tests for now:

```
1) Test_VObjects::testCrappyVCard
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'Fermée;Adèle;;;'
+'Fermée\;Adèle\;\;\;;;;;'

/srv/www/htdocs/owncloud/apps3/contacts/tests/lib/vobject_test.php:25

2) Test_VObjects::testEscapedParameters
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'Parameters;Escaped;;;'
+'Parameters\;Escaped\;\;\;;;;;'

/srv/www/htdocs/owncloud/apps3/contacts/tests/lib/vobject_test.php:37
```

**Comment by PVince81** [comment]
- [x] CardDAV broken again: a lot of static init code doesn't like IUserSession... or runs too early

**Comment by PVince81** [comment]
So... added back `\OCP:\User->getUser()` in two places... not sure how this has ever worked before because the user might not even be initialized at that point... Anyway, that makes CardDAV work a bit.

**Comment by PVince81** [comment]
This is ready to go. It's not the perfect fix but the goal was to make the contacts app work with the new Sabre in a short time frame, which this PR achieves.

Please review.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 language: php
 php:
-  - 5.3
```
Comment (nickvergessen) [inline .travis.yml]: Minimum requirement changed? If so, put it into the appinfo?

Comment (PVince81) [inline .travis.yml]: done

**Comment by PVince81** [comment]
Sabre was merged in master for OC 8.1, please review this PR.

**Comment by nickvergessen** [comment]
Reopened for new test run

**Comment by LEDfan** [comment]
I'll give this is a good test today. A quick overview for what to test:
- [ ] contacts sync (e.g. thunderbird, android)
  - [ ] remove/add//edit
- [ ] shared, database and ldap contacts backend must be tested

**Comment by LEDfan** [comment]
This currently doesn't work for me. When I add the url in Thunderbird, there is no authentication popup. When I manually go to the url I first get this error:

```
( ! ) Fatal error: Call to a member function log() on a non-object in /var/www/owncloud/lib/private/connector/sabre/exceptionloggerplugin.php on line 88
```

Next when I var_export the error on line 88 instead of trying to log it I get:

```
array ( 'Message' => 'HTTP/1.1 403 User did not have the required privileges ({DAV:}read) for path "addressbooks/admin"', 'Code' => 0, 'Trace' => '#0 /var/www/owncloud/3rdparty/sabre/dav/lib/DAVACL/Plugin.php(769): Sabre\\DAVACL\\Plugin->checkPrivileges(\'addressbooks/ad...\', \'{DAV:}read\') #1 [internal function]: Sabre\\DAVACL\\Plugin->beforeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response)) #2 /var/www/owncloud/3rdparty/sabre/event/lib/EventEmitterTrait.php(105): call_user_func_array(Array, Array) #3 /var/www/owncloud/3rdparty/sabre/dav/lib/DAV/Server.php(456): Sabre\\Event\\EventEmitter->emit(\'beforeMethod\', Array) #4 /var/www/owncloud/3rdparty/sabre/dav/lib/DAV/Server.php(254): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response)) #5 /var/www/owncloud/apps/contacts/appinfo/remote.php(83): Sabre\\DAV\\Server->exec() #6 /var/www/owncloud/remote.php(58): require_once(\'/var/www/ownclo...\') #7 {main}', 'File' => '/var/www/owncloud/3rdparty/sabre/dav/lib/DAVACL/Plugin.php', 'Line' => 223, ) 2.1.2 Sabre\DAVACL\Exception\NeedPrivileges User did not have the required privileges ({DAV:}read) for path "addressbooks/admin" /remote.php/carddav/addressbooks/admin 
```

In crhome I'm getting:

```
array ( 'Message' => 'HTTP/1.1 401 No basic authentication headers were found', 'Code' => 0, 'Trace' => '#0 /var/www/owncloud/lib/private/connector/sabre/auth.php(126): Sabre\\DAV\\Auth\\Backend\\AbstractBasic->authenticate(Object(Sabre\\DAV\\Server), \'ownCloud\') #1 /var/www/owncloud/lib/private/connector/sabre/auth.php(106): OC\\Connector\\Sabre\\Auth->auth(Object(Sabre\\DAV\\Server), \'ownCloud\') #2 /var/www/owncloud/3rdparty/sabre/dav/lib/DAV/Auth/Plugin.php(118): OC\\Connector\\Sabre\\Auth->authenticate(Object(Sabre\\DAV\\Server), \'ownCloud\') #3 [internal function]: Sabre\\DAV\\Auth\\Plugin->beforeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response)) #4 /var/www/owncloud/3rdparty/sabre/event/lib/EventEmitterTrait.php(105): call_user_func_array(Array, Array) #5 /var/www/owncloud/3rdparty/sabre/dav/lib/DAV/Server.php(456): Sabre\\Event\\EventEmitter->emit(\'beforeMethod\', Array) #6 /var/www/owncloud/3rdparty/sabre/dav/lib/DAV/Server.php(254): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response)) #7 /var/www/owncloud/apps/contacts/appinfo/remote.php(83): Sabre\\DAV\\Server->exec() #8 /var/www/owncloud/remote.php(58): require_once(\'/var/www/ownclo...\') #9 {main}', 'File' => '/var/www/owncloud/3rdparty/sabre/dav/lib/DAV/Auth/Backend/AbstractBasic.php', 'Line' => 70, ) 2.1.2 Sabre\DAV\Exception\NotAuthenticated No basic authentication headers were found
```

**Comment by LEDfan** [comment]
Mm. webdav is also broken on my test machine. I'll rebuild my vm and report back.

**Comment by PVince81** [comment]
Thanks @LEDfan. Thing is I didn't test syncing and only did some quick curl WebDAV calls.
So it is likely that it will need fixing too.

**Comment by LEDfan** [comment]
@PVince81 I tried again with a new VM. same troubles.

**Comment by PVince81** [comment]
Ok thanks for testing. I'll have to find some time to fix it then, unless you want to give it a go ? (feel free to push on this branch)

**Comment by PVince81** [comment]
@LEDfan are these shared address books ?
I tried again with regular address books and I'm able to access with curl.

Can you try this:
`curl -X PROPFIND http://root:admin@localhost/owncloud/remote.php/carddav/addressbooks/root/ | xmllint --format -` (replace "root" with your user name)

If it doesn't work, try again with the parent folder, etc.
In my case it just returns correct XML.

**Comment by PVince81** [comment]
Squashed/rebased

**Comment by PVince81** [comment]
I still suggest to merge this. Without this the contacts app is completely unusable due to API issues.
If you can provide me with more details about your CardDAV issue I can try and fix it.

**Comment by LEDfan** [comment]
Indeed let's merge this, I'll test with curl later today.

**Comment by LEDfan** [comment]
@PVince81 both Thunderbird (add,delete,edit) and curl are working fine now! Thanks.

**Comment by derkostka** [comment]
I just received this error when syncing on owncloud 8.1 alpha + contacts head from today ..

15 of 117 contacts were synced to my android devce, then this error was shown (also on the android device by the way)

{"reqId":"VUkPmH8AAAEAAEuAEvwAAAAA","remoteAddr":"2001:4ba0:ffec:1aa::1","app":"caldav","message":"Exception: {\"Message\":\"HTTP\\/1.1 403 User did not have the required privileges ({DAV:}read) for path \\"calendars\\/sebastian.kostka@gmail.com\\/contact_birthdays\\/local::15::4051.ics\\"\",\"Code\":0,\"Trace\":\"#0 \\/var\\/www\\/owncloud\\/3rdparty\\/sabre\\/dav\\/lib\\/DAVACL\\/Plugin.php(769): Sabre\\DAVACL\\Plugin->checkPrivileges('calendars\\/sebas...', '{DAV:}read')\n#1 [internal function]: Sabre\\DAVACL\\Plugin->beforeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#2 \\/var\\/www\\/owncloud\\/3rdparty\\/sabre\\/event\\/lib\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#3 \\/var\\/www\\/owncloud\\/3rdparty\\/sabre\\/dav\\/lib\\/DAV\\/Server.php(456): Sabre\\Event\\EventEmitter->emit('beforeMethod', Array)\n#4 \\/var\\/www\\/owncloud\\/3rdparty\\/sabre\\/dav\\/lib\\/DAV\\/Server.php(254): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#5 \\/var\\/www\\/owncloud\\/apps2\\/calendar\\/appinfo\\/remote.php(57): Sabre\\DAV\\Server->exec()\n#6 \\/var\\/www\\/owncloud\\/remote.php(84): require_once('\\/var\\/www\\/ownclo...')\n#7 {main}\",\"File\":\"\\/var\\/www\\/owncloud\\/3rdparty\\/sabre\\/dav\\/lib\\/DAVACL\\/Plugin.php\",\"Line\":223}","level":4,"time":"May 05, 2015 20:44:40"}

**Pull request merged**